File: samples/cpp/tutorial_code/core/AddingImages/AddingImages.cpp
/**
 * @file AddingImages.cpp
 * @brief Simple linear blender ( dst = alpha*src1 + beta*src2 )
 * @author OpenCV team
 */
#include "opencv2/imgcodecs.hpp"
#include "opencv2/highgui.hpp"
#include <iostream>

using namespace cv;

// we're NOT "using namespace std;" here, to avoid collisions between the beta variable and std::beta in c++17
using std::cin;
using std::cout;
using std::endl;

/**
 * @function main
 * @brief Main function
 */
int main( void )
{
   double alpha = 0.5; double beta; double input;

   Mat src1, src2, dst;

   /// Ask the user enter alpha
   cout << " Simple Linear Blender " << endl;
   cout << "-----------------------" << endl;
   cout << "* Enter alpha [0.0-1.0]: ";
   cin >> input;

   // We use the alpha provided by the user if it is between 0 and 1
   if( input >= 0 && input <= 1 )
     { alpha = input; }

   //![load]
   /// Read images ( both have to be of the same size and type )
   src1 = imread( samples::findFile("LinuxLogo.jpg") );
   src2 = imread( samples::findFile("WindowsLogo.jpg") );
   //![load]

   if( src1.empty() ) { cout << "Error loading src1" << endl; return EXIT_FAILURE; }
   if( src2.empty() ) { cout << "Error loading src2" << endl; return EXIT_FAILURE; }

   //![blend_images]
   beta = ( 1.0 - alpha );
   addWeighted( src1, alpha, src2, beta, 0.0, dst);
   //![blend_images]

   //![display]
   imshow( "Linear Blend", dst );
   waitKey(0);
   //![display]

   return 0;
}

File: samples/python/tutorial_code/core/AddingImages/adding_images.py
from __future__ import print_function
import sys

import cv2 as cv

alpha = 0.5

print(''' Simple Linear Blender
-----------------------
* Enter alpha [0.0-1.0]: ''')
if sys.version_info >= (3, 0): # If Python 3.x
    input_alpha = float(input())
else:
    input_alpha = float(raw_input())
if 0 <= alpha <= 1:
    alpha = input_alpha
## [load]
src1 = cv.imread('../../../../data/LinuxLogo.jpg')
src2 = cv.imread('../../../../data/WindowsLogo.jpg')
## [load]
if src1 is None:
    print ("Error loading src1")
    exit(-1)
elif src2 is None:
    print ("Error loading src2")
    exit(-1)
## [blend_images]
beta = (1.0 - alpha)
dst = cv.addWeighted(src1, alpha, src2, beta, 0.0)
## [blend_images]
## [display]
cv.imshow('dst', dst)
cv.waitKey(0)
## [display]
cv.destroyAllWindows()
